src/core/validations/message.rs

```diff
@@ -208,6 +208,9 @@ pub fn validate_message(
             validate_frame_action_body(&frame_action_body)?;
         }
         Some(proto::message_data::Body::LendStorageBody(lend_storage_body)) => {
+            if message_data.timestamp < block_timestamp.decr_by(60 * 60).to_u64() as u32 {
```

Comment:
Leaning toward making this window even smaller. Maybe something like 10m just to account for clock skew and minor delays.

Reply:
Sounds good. I think it's easier to keep it more restrictive and relax later. Normal users won't hit this, so should be fine to be aggressive.